geode-core/src/main/java/org/apache/geode/internal/cache/partitioned/ClearPRMessage.java

```diff
@@ -0,0 +1,405 @@
+import org.apache.logging.log4j.Logger;
+
+import org.apache.geode.DataSerializer;
+import org.apache.geode.cache.CacheException;
+import org.apache.geode.cache.EntryExistsException;
+import org.apache.geode.distributed.DistributedLockService;
+import org.apache.geode.distributed.DistributedMember;
+import org.apache.geode.distributed.internal.ClusterDistributionManager;
+import org.apache.geode.distributed.internal.DirectReplyProcessor;
+import org.apache.geode.distributed.internal.DistributionManager;
+import org.apache.geode.distributed.internal.InternalDistributedSystem;
+import org.apache.geode.distributed.internal.ReplyException;
+import org.apache.geode.distributed.internal.ReplyMessage;
+import org.apache.geode.distributed.internal.ReplyProcessor21;
+import org.apache.geode.distributed.internal.ReplySender;
+import org.apache.geode.distributed.internal.membership.InternalDistributedMember;
+import org.apache.geode.internal.Assert;
+import org.apache.geode.internal.InternalDataSerializer;
+import org.apache.geode.internal.NanoTimer;
+import org.apache.geode.internal.cache.BucketRegion;
+import org.apache.geode.internal.cache.DataLocationException;
+import org.apache.geode.internal.cache.EventID;
+import org.apache.geode.internal.cache.ForceReattemptException;
+import org.apache.geode.internal.cache.PartitionedRegion;
+import org.apache.geode.internal.cache.PartitionedRegionException;
+import org.apache.geode.internal.cache.PartitionedRegionHelper;
+import org.apache.geode.internal.cache.RegionEventImpl;
+import org.apache.geode.internal.logging.log4j.LogMarker;
+import org.apache.geode.internal.serialization.DeserializationContext;
+import org.apache.geode.internal.serialization.SerializationContext;
+import org.apache.geode.logging.internal.log4j.api.LogService;
+
+public class ClearPRMessage extends PartitionMessageWithDirectReply {
+  private static final Logger logger = LogService.getLogger();
+
+  private RegionEventImpl regionEvent;
+
+  private Integer bucketId;
+
+  private Object callbackArg;
+
+  /** The time in ms to wait for a lock to be obtained during doLocalClear() */
+  public static final int LOCK_WAIT_TIMEOUT_MS = 100;
+  public static final String BUCKET_NON_PRIMARY_MESSAGE =
+      "The bucket region on target member is no longer primary";
+  public static final String BUCKET_REGION_LOCK_UNAVAILABLE_MESSAGE =
+      "A lock for the bucket region could not be obtained.";
+
+  protected static final short HAS_BRIDGE_CONTEXT = UNRESERVED_FLAGS_START;
+
+  /**
+   * state from operateOnRegion that must be preserved for transmission from the waiting pool
+   */
+  transient boolean result = false;
+
+  /**
+   * Empty constructor to satisfy {@link DataSerializer}requirements
+   */
+  public ClearPRMessage() {}
+
+  public ClearPRMessage(int bucketId, boolean notificationOnly, boolean posDup,
+      Object callbackArg) {
+    this.bucketId = bucketId;
+    this.notificationOnly = notificationOnly;
+    this.posDup = posDup;
```

Comment:
I don't think we should remove this entirely as they are fields of the parent class and are called by methods in the parent class that will end up running when this message is sent - however, as we never need this to be anything but false, I have changed it to automatically set this value to false in the constructer, rather than allowing a custom value for each instance of the message via the constructor.

Reply:
Since they're `boolean` you don't need to explicitly set them to false; they default to that value even if they're not initialized, so we can avoid cluttering the constructor with assignments that are never used by the class.

Reply:
This is true, but I feel expressly setting them is more clear. If they were expressly set to false in the parent class as a default that would be different, but just allowing the compiler to default their values to false kind of hides the fact that this is an intentional value. Our message class will use these values (albeit in methods defined by its parent) and it's important that they're set to false, otherwise we run the risk of improperly processing the message.